docs/blogs/2023/2023-06-07-fast-conformer.md

```diff
@@ -0,0 +1,210 @@
+
+
+*Below, we tabulate the accuracy vs speed for each component of Fast Conformer modification schema. Models were tested on LibriSpeech test-other incrementally for each modification starting from the original Conformer. Encoder inference speed (samples/sec) was measured with batch size 128 on 20 sec audio samples. The number of parameters (M) is shown for the encoder only.* 
+
+| Encoder             	| WER, Test Other % 	| Inference Samples / sec 	| Params (M) 	| GMACS 	|
+|---------------------	|-------------------	|-------------------------	|------------	|-------	|
+| Baseline Conformer  	| 5.19              	| 169                     	| 115        	| 143.2 	|
+| \  +8X Stride       	| 5.11              	| 303                     	| 115        	| 92.5  	|
+| \ \ +Depthwise conv 	| 5.12              	| 344                     	| 111        	| 53.2  	|
+| \ \ \ +256 channels 	| 5.09              	| 397                     	| 109        	| 48.8  	|
+| \ \ \ \ +Kernel 9   	| 4.99              	| 467                     	| 109        	| 48.7  	|
+
+
+## Fast Conformer: Linearly Scalable Attention
+
+On an NVIDIA A100 GPU with 80 GB of memory, we find that Conformer reaches the memory limit at around 10-minute long single audio clip. This means that for it is not feasible to perform inference without performing streaming inference which may lead to degraded results.
+
+Fast Conformer, due to its 8x stride fairs a little better and can perform inference on roughly 18-minute long audio clips. However, this is still not sufficient for many use cases.
+
+We therefore extend [Longformer](https://arxiv.org/abs/2004.05150) to the Conformer architecture. Longformer uses local attention augmented with global tokens.  We use a single global attention token, which attends to all other tokens and has all other tokens attend to it, using a separate set of query, key and value linear projections, while others attend in a fixed-size window surrounding each token (see below). 
+
+<div align="center">
+<img src="https://github.com/NVIDIA/NeMo/releases/download/v1.18.0/asset-post-fast-conformer-local-attn.png" width="100%">
+</div>
+
+By switching to limited context attention, we extend the maximum duration that the model can process at once on a single A100 GPU by 7x: from 10 minutes for Conformer to 70 minutes for Fast Conformer. Furthermore, you can use a pre-trained Fast Conformer model and readily convert its attention to Longformer attention without any further training ! While this will not use the global attention token, it will still be able to process 70-minute long audio clips.
```

Comment:
by 4x since we already got to 18 min by down sampling the channels in FC, then long former to get to 70.

Reply:
Very good catch